plaidml2/edsl/edsl_test.py

```diff
@@ -50,6 +67,44 @@ def conv_2d(I, K):
     R[n, x0, x1, co] += I[n, x0 + k0 - (K0 // 2), x1 + k1 - (K1 // 2), ci] * K[k0, k1, ci, co]
     return R
 
+def complex_conv_2d(
+    I,
+    K,
+    s0,s1,  # stride coeffs
+    d0,d1   # dilation coeffs
+) :
+    # "same-lower" autopadding will be applied
+    N, G, GCI, GCO = TensorDims(4)
+    X0, X1 = TensorDims(2)
+    K0, K1 = TensorDims(2)
+    n, g, gci, gco = TensorIndexes(4)
+    x0, x1 = TensorIndexes(2)
+    k0, k1 = TensorIndexes(2)
+    I.bind_dims(N, X0, X1, G, GCI)
+    K.bind_dims(K0, K1, G, GCI, GCO)
+
+    # Compute output spatial dimensions
+    Y0, Y1 = TensorDims(2)
+    Y0 = (X0 + s0 - 1) // s0
+    Y1 = (X1 + s1 - 1) // s1
+
+    #Compute the effective kernel size after dilation
+    EK0, EK1 = TensorDims(2)
+    EK0 = d0 * (K0 - 1) + 1
+    EK1 = d1 * (K1 - 1) + 1
+
+   #Compute the padding offset
```

Comment:
Run `yapf` to fix formatting

Reply:
I used yapf. It didn't delete the blank lines, are those the problem? They were for better readability in the document. I can remove them manually if needed.